src/game/Tactical/Points.cc

```diff
@@ -1116,18 +1116,40 @@ static UINT8 MinAPsToPunch(SOLDIERTYPE const& s, GridNo gridno, bool const add_t
 		{
 			ap += GetAPsToChangeStance(&s, ANIM_CROUCH);
 		}
-		else if (s.sGridNo == gridno)
+		else
 		{
 			ap += GetAPsToChangeStance(&s, ANIM_STAND);
 		}
```

Comment:
are we missing a case here, when there's no need for a stance change?

Reply:
Only lying/proning enemys need to be attacked while crouching for every other case the merc would need to stand. If there is no need for a stance change 0 will be returned by the GetAPsToChangeStance function. 

I don't know what the else if statement 
`else if (s.sGridNo == gridno)`
was for because the only case in which your merc can have the same gridno like your enemy is when they're on different height levels(roof/ground). But even this wouldn't be possible as you would receive an error message.

Reply:
Which reminds me, why is there a special variable for gridno in the outer block? In other words, is tgt->sGridNo ever different from the gridno passed to the function? The way the target is derived, I'd hope it wouldn't be possible.